cmake/optional.cmake

```diff
@@ -111,9 +111,13 @@ endif()
 
 if(ENABLE_WALLET)
   if(WITH_SQLITE)
-    include(CrossPkgConfig)
-    cross_pkg_check_modules(sqlite sqlite3>=3.7.17 IMPORTED_TARGET)
-    if(sqlite_FOUND)
+    if(MSVC)
+      find_package(unofficial-sqlite3 CONFIG)
```

Comment:
Could you please explain (in the form of a comment :) what "unofficial-sqlite3" is and where it comes from and why we care?

Reply:
It is just a name of vcpkg's package:
```
C:\>vcpkg --triplet=x64-windows-static install sqlite3
Computing installation plan...
The following packages are already installed:
    sqlite3[core,json1]:x64-windows-static -> 3.42.0
sqlite3:x64-windows-static is already installed
Total install time: 293 us
sqlite3 provides pkgconfig bindings.
sqlite3 provides CMake targets:

    find_package(unofficial-sqlite3 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::sqlite3::sqlite3)

```

Reply:
What wording do you prefer for the comment?

Reply:
I guess my question is... is there an official one? If this is not intended to be the canonical release, what is?

Reply:
> I guess my question is... is there an official one? If this is not intended to be the canonical release, what is?

The reason of the `unofficial::` namespace is:
> vcpkg owns the CMake build of sqlite3 and exports the targets via a CMakeLists.txt file in the port folder 

I understand the sentence above as SQLite project does not support CMake [for now](https://github.com/sqlite/sqlite/pull/10).

And it is _only_ a namespace.

Btw, we are currently using the same: https://github.com/hebasto/bitcoin/blob/d09c8bc730d8d412ddc9b040cbeeb49dff3104de/build_msvc/vcpkg.json#L10

Reply:
An important note. This PR does not changes the SQLite dependency. It changes the way how the build system discovers it.

Reply:
Ok, thanks for explaining. It sure seems that "vcpkg" would've been a better namespace, heh. It's odd that it's named for what it's not rather than what it is.

I suppose just a comment to the effect of "Use of the `unofficial` namespace is a vcpkg convention" would be helpful.

Reply:
> > I guess my question is... is there an official one? If this is not intended to be the canonical release, what is?
> 
> The reason of the `unofficial::` namespace is:
> 
> > vcpkg owns the CMake build of sqlite3 and exports the targets via a CMakeLists.txt file in the port folder
> 
> I understand the sentence above as the SQLite project does not support CMake [for now](https://github.com/sqlite/sqlite/pull/10).

I have to amend my comment.

From vcpkg's [Maintainer Guide](https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#add-cmake-exports-in-an-unofficial--namespace):
> A core design ideal of vcpkg is to not create "lock-in" for customers. In the build system, there should be no difference between depending on a library from the system, and depending on a library from vcpkg. To that end, we avoid adding CMake exports or targets to existing libraries with "the obvious name", to allow upstreams to add their own official CMake exports without conflicting with vcpkg.
>
> To that end, any CMake configs that the port exports, which are not in the upstream library, should have `unofficial-` as a prefix. Any additional targets should be in the `unofficial::<port>::` namespace.